java/core/src/main/java/com/whylogs/api/logger/Logger.java

```diff
@@ -1,3 +1,85 @@
 package com.whylogs.api.logger;
 
-public class Logger {}
+import com.whylogs.api.logger.resultsets.ProfileResultSet;
+import com.whylogs.api.logger.resultsets.ResultSet;
+import com.whylogs.api.writer.Writer;
+import com.whylogs.api.writer.WritersRegistry;
+import com.whylogs.core.DatasetProfile;
+import com.whylogs.core.schemas.DatasetSchema;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+
+@NoArgsConstructor
+@Getter
+@EqualsAndHashCode
+@ToString
+public abstract class Logger implements AutoCloseable {
+  private boolean isClosed = false;
+  private DatasetSchema schema = new DatasetSchema();
+  private ArrayList<Writer> writers = new ArrayList<>();
+
+  public Logger(DatasetSchema schema) {
+    this.schema = schema;
+  }
+
+  public <T extends Writer> void checkWriter(T Writer) {
+    // Checks if a writer is configured correctly for this class
+    // Question: why is this empty but not an abstract?
+  }
+
+  public void appendWriter(String name) {
+    if (name == null || name.isEmpty()) {
+      throw new IllegalArgumentException("Writer name cannot be empty");
+    }
+
+    Writer writer = WritersRegistry.get(name);
+    if (writer == null) {
+      throw new IllegalArgumentException("Writer " + name + " is not registered");
+    }
+
+    appendWriter(writer);
+  }
+
+  public void appendWriter(Writer writer) {
+    if (writer == null) {
+      throw new IllegalArgumentException("Writer cannot be null");
+    }
+
+    checkWriter(writer);
+    writers.add(writer);
+  }
+
+  protected abstract ArrayList<DatasetProfile> getMatchingProfiles(Object data);
+
+  protected abstract <O> ArrayList<DatasetProfile> getMatchingProfiles(Map<String, O> data);
+
+  @Override
+  public void close() {
+    isClosed = true;
+  }
+
+  public ResultSet log(HashMap<String, Object> data) {
```

Comment:
I assume this corresponds to the Python `log(row=...)` and the other data formats will come later?

Reply:
Correct :)